Angular/Angular guidelines and best practices/DatX data store.md

```diff
@@ -0,0 +1,206 @@
+  ) {}
+}
+```
+
+When testing consumers of APP_COLLECTION, just inject the collection under the same token again as with any other dependency:
+
+```ts
+// example.service.spec.ts
+import { AppCollection } from '<path-to-collection-definition>';
+import { APP_COLLECTION } from '<path-to-token-definition>';
+import { TestBed } from '@angular/core/testing';
+import { ExampleService } from './example.service';
+
+describe('ExampleService', () => {
+	let service: ExampleService;
+
+	beforeEach(() => {
+		TestBed.configureTestingModule({
+			providers: [
+				{
+					provide: APP_COLLECTION,
+					useValue: new AppCollection(), // empty mock collection
+				},
+			],
+		});
+		service = TestBed.inject(ExampleService);
+	});
+
+	it('should be created', () => {
+		expect(service).toBeTruthy();
+	});
+});
+```
+
+## Configuring DatX HTTP calls interception
+
+By default, DatX uses Fetch API when invoking HTTP calls, but this means that it also bypass **HttpClient** from `@angular/common/http` and therefore any registered interception logic or similar kind of middleware. Luckily there is a way around that, see [official documentation guide](https://datx.dev/docs/jsonapi-angular/base-fetch).
+
+## Abstracting entity related login into separate service
+
+You might want to abstract away a little of tediousness of working with DatX and/or bound some special logic to a given entity. Since many of the Collection's method require an Entity type as a parameter you can create separate service that will hide this small implementation detail for you. One way to do that would be to extend an abstract class similar to this one:
+
+```ts
+// collection.service.ts
+import { Inject } from '@angular/core';
+import { IModelConstructor, IRawModel, IType } from '@datx/core';
+import { IRequestOptions } from '@datx/jsonapi';
+import { IJsonapiModel, Response } from '@datx/jsonapi-angular';
+import { Observable } from 'rxjs';
+import { map, mapTo } from 'rxjs/operators';
+import { AppCollection } from '<path-to-collection-definition>';
+import { APP_COLLECTION } from '<path-to-token-definition>';
+
+export abstract class CollectionService<TModel extends IJsonapiModel> {
+	protected abstract readonly ctor: IModelConstructor<TModel>;
+
+	constructor(@Inject(APP_COLLECTION) protected readonly collection: AppCollection) {}
+
+	public create(rawModel: IRawModel | Record<string, unknown>): TModel {
+		if (rawModel.id === null || rawModel.id === undefined || rawModel.id === '') {
+			delete rawModel.id;
+		}
+
+		return this.collection.add(rawModel, this.ctor);
+	}
+
+	public createAndSave(rawModel: IRawModel | Record<string, unknown>): Observable<TModel> {
+		const model = this.create(rawModel);
+		return this.save(model);
+	}
+
+	public findAll(): Array<TModel> {
+		return this.collection.findAll<TModel>(this.ctor);
+	}
+
+	public getAllModels(options?: IRequestOptions): Observable<Array<TModel>> {
+		return this.getMany({
```

Comment:
What does this exactly do? Seems to me like it might be something project specific?

Reply:
Yes, it's optional, wording fits "You might want to abstract away a little of the tediousness of working with DatX and/or bound some special logic to a given entity.". Do you want me to remove it?

Reply:
No, it can stay, but I'm not sure what does the `getAllModels` represent - I think we're missing a better example if we want to show custom logic for a single entity.

Reply:
Added comment explaining that in 7dc1020